test/e2e/config/vsphere-dev.yaml

```diff
@@ -0,0 +1,180 @@
+      value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.1/control-plane-components.yaml"
+      type: "url"
+      contract: v1beta1
+      replacements:
+        - old: "imagePullPolicy: Always"
+          new: "imagePullPolicy: IfNotPresent"
+      files:
+      - sourcePath: "../data/shared/metadata.yaml"
+
+  - name: vsphere
+    type: InfrastructureProvider
+    versions:
+      - name: v0.7.10
+        value: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v0.7.10/infrastructure-components.yaml
+        type: url
+        contract: v1alpha3
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+        files:
+          # TODO: v1a3 cluster-template includes WORKLOAD_CONTROL_PLANE_ENDPOINT_IP
+          - sourcePath: "../data/shared/vsphere/metadata.yaml"
+          - sourcePath: "../data/infrastructure-vsphere/capi-upgrades/v1alpha3/cluster-template.yaml"
+      - name: v0.8.1
+        type: url
+        contract: v1alpha4
+        value: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v0.8.1/infrastructure-components.yaml
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+        files:
+          - sourcePath: "../data/shared/vsphere/metadata.yaml"
+          - sourcePath: "../data/infrastructure-vsphere/capi-upgrades/v1alpha4/cluster-template.yaml"
+      - name: v1.0.1
+        type: url
+        contract: v1beta1
+        value: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v1.0.1/infrastructure-components.yaml
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+        files:
+          - sourcePath: "../data/shared/vsphere/metadata.yaml"
+          - sourcePath: "../data/infrastructure-vsphere/capi-upgrades/v1beta1/cluster-template.yaml"
+
+variables:
+  KUBERNETES_VERSION: "v1.22.3"
+  CNI: "./data/cni/calico/calico.yaml"
+  EXP_CLUSTER_RESOURCE_SET: "true"
+  CONTROL_PLANE_MACHINE_COUNT: 1
+  WORKER_MACHINE_COUNT: 1
+  IP_FAMILY: "IPv4"
+  # Following CAPV variables should be set before testing
+  VSPHERE_SERVER: "vcenter.vmware.com"
+  VSPHERE_DATACENTER: "dc0"
+  VSPHERE_DATASTORE: "WorkloadDatastore"
+  VSPHERE_STORAGE_POLICY: "Cluster API vSphere Storage Policy"
+  VSPHERE_FOLDER: "rp0"
+  VSPHERE_NETWORK: "VM Network"
+  VSPHERE_RESOURCE_POOL: "ResourcePool"
+  VSPHERE_TEMPLATE: "ubuntu-2004-kube-v1.21.2"
+  # Also following variables are required but it is recommended to use env variables to avoid disclosure of sensitive data
+  # VSPHERE_SSH_AUTHORIZED_KEY:
+  #  VSPHERE_USERNAME:
+  #  VSPHERE_PASSWORD:
+
+intervals:
+  default/wait-controllers: ["5m", "10s"]
+  default/wait-cluster: ["5m", "10s"]
+  default/wait-control-plane: ["10m", "10s"]
```

Comment:
I saw on the vsphere master branch, it's set to 20m by default. is there any reason we want to change that?
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/test/e2e/config/vsphere-dev.yaml#L190-L191

Reply:
no, there is no specific reason. we can set it to be the same as CAPV.

Reply:
Let's change it back to what upstream use